### PR TITLE
remove `new` from `yarn create comit-app new foo`

### PR DESCRIPTION
**Pull request by D4nte**
The interaction now looks like that:

```
~/src/temp  master ✗                                                                                                                                                                                    ◒
▶ yarn create comit-app bla
yarn create v1.19.2
[1/4] 🔍  Resolving packages...
[2/4] 🚚  Fetching packages...
[3/4] 🔗  Linking dependencies...
warning " > tscodeshift@0.4.1" has incorrect peer dependency "typescript@^2.3.2".
warning "tscodeshift > ts-emitter@0.3.1" has incorrect peer dependency "typescript@^2.3.2".
[4/4] 🔨  Building fresh packages...
success Installed "create-comit-app@0.6.0-rc1" with binaries:
      - create-comit-app
      - comit-scripts
Your project `bla` has been created!
Now, start your development environment by running `create-comit-app start-env`
✨  Done in 3.12s.

~/src/temp  master ✗                                                                                                                                                                                    ◒
▶ ls bla
README.md    examples     package.json src

~/src/temp  master ✗                                                                                                                                                                                    ◒
▶ comit-scripts start-env
Creating Docker network (create-comit-app)...✓
Starting Ethereum node...✓
Starting Bitcoin node...✓
Starting two cnds...✓
Writing configuration in env file...✓
🎉 Environment is ready, time to create a COMIT app!
```

### CODE REVIEW

File: .npm/execute.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
+const spawn = require("child_process").spawn;
+
+export async function execute(binPath, args) {
+  const cca = spawn(binPath, args);
+
+  cca.on("error", error => {
+    console.error("Could not execute create-comit-app:", error);
+  });
+
+  async function handleSignal(code) {
```
Comment (thomaseizinger) [inline .npm/execute.js]: This seems to be unnecessarily `async`, you are not `await`-ing in this function.

File: .npm/execute.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    cca.kill(code);
+  }
+
+  process.on("beforeExit", async code => {
```
Comment (thomaseizinger) [inline .npm/execute.js]: I am not sure if nodejs supports `async` signal handlers: https://github.com/nodejs/node/issues/29480

File: .npm/new.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+const { execute } = require("./execute");
+const path = require("path");
+
+(async () => {
+  let args = process.argv;
+  args.shift(); // node
+  const dirPath = path.dirname(args[0]);
+  args[0] = "new"; // Replace ".../cca.js" with "new"
+
+  const ccaVersion = /^\d\.\d\.\d/.exec(packageJson.version)[0];
```
Comment (thomaseizinger) [inline .npm/new.js]: What is the reason for using a regex here? The version number in the package.json is exactly `0.6.0`. Is that not sufficient?

File: .npm/yarn-error.log [hunk 4]
```diff
@@ -0,0 +1,275 @@
+Arguments:
```
Comment (thomaseizinger) [inline .npm/yarn-error.log]: This file looks like it should not be part of the patch :)

File: new_project/examples/btc_eth/package.json [hunk 5]
```diff
@@ -7,7 +7,7 @@
         "check": "tsc && tslint -p . && prettier --check '**/*.{ts,json,yml}'",
         "swap": "ts-node ./src/index.ts",
         "fix": "tslint -p . --fix && prettier --write '**/*.{ts,js,json,yml}'",
-        "start-env": "create-comit-app start-env"
```
Comment (thomaseizinger) [inline new_project/examples/btc_eth/package.json]: I am not (yet) sure if I like that :thinking: 

It looks very magical to me because the link between `create-comit-app` and `start-env` is now completely missing. It is an old discussion but what if we just follow the create-react-app path and provide a `comit-scripts` binary.

I didn't realize it before but it looks like we can provide several NPM "binaries" as part of one package? That is cool! It would be cool if this would read `comit-scripts start-env`

File: .npm/new.js [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+  const binPath = `${dirPath}/create-comit-app_${ccaVersion}/create-comit-app`;
```
Comment (thomaseizinger) [inline .npm/new.js]: This is still not going to work on windows (.exe is missing). Would be cool to fix this as part of this PR as-well!

File: .npm/execute.js (hunk 1 above)
Comment (D4nte) [reply .npm/execute.js]: Yes we are line 15, 19, 23.

File: .npm/execute.js (hunk 2 above)
Comment (D4nte) [reply .npm/execute.js]: Ok, I can play around with it but if it is here in the first place I think it's because I had issue without it (this is not new code).

File: .npm/new.js (hunk 3 above)
Comment (D4nte) [reply .npm/new.js]: For this exact kind of PR I will do manual publication on npm with RC versions so that I can test the PR work before doing a full create-comit-app release. Hence the package.json will be `0.6.0rc1`, hence we need to extract.

File: new_project/examples/btc_eth/package.json (hunk 5 above)
Comment (D4nte) [reply new_project/examples/btc_eth/package.json]: I see. I think we roll with it (maybe I'll choose a better name) and restructure to comit-scripts at a later stage as what you want would most likely mean we publish 2 npm packages: `create-comit-app` and `comit-scripts`. Does this mean they both make you download create-comit-app binary ?

File: .npm/new.js (hunk 6 above)
Comment (D4nte) [reply .npm/new.js]: Can we please agree on the windows strategy first?

File: new_project/examples/btc_eth/package.json (hunk 5 above)
Comment (D4nte) [reply new_project/examples/btc_eth/package.json]: Ok I am going to try something :)

Comment (D4nte) [reply new_project/examples/btc_eth/package.json]: See description, I did what I think you expect.

File: .npm/execute.js (hunk 1 above)
Comment (thomaseizinger) [reply .npm/execute.js]: I was taking about _this_ function. The function `handleSignal` is one line long and it does not use `await`. Hence the `async` is unnecessary. Did you maybe mean to _return_ the return value of `cca.kill`?

Comment (thomaseizinger) [reply .npm/execute.js]: `kill` does not return a Promise: https://nodejs.org/api/child_process.html#child_process_subprocess_kill_signal

I think you can safely remove the `async` here :)

File: .npm/execute.js (hunk 2 above)
Comment (thomaseizinger) [reply .npm/execute.js]: All of these `await` can go away anyway given [the above comment](https://github.com/comit-network/create-comit-app/pull/280/files/5d2f9126f2e8844b807752171942c169fd54564d#r356565601).

File: .npm/new.js (hunk 3 above)
Comment (thomaseizinger) [reply .npm/new.js]: Makes perfect sense :)
Do you mind documenting that in a commend / function? It was not obvious to me!

File: .npm/package.json [hunk 7]
```diff
@@ -25,7 +26,8 @@
   },
   "homepage": "https://github.com/comit-network/create-comit-app#readme",
   "bin": {
-    "create-comit-app": "cca.js"
+    "create-comit-app": "new.js",
+    "comit-scripts": "scripts.js"
```
Comment (thomaseizinger) [inline .npm/package.json]: :heart_eyes:

File: .npm/yarn-error.log (hunk 4 above)
[the remark this answers is not in the document]
Comment (thomaseizinger) [reply .npm/yarn-error.log]: No but it is also removed in the latest diff :)
